File: Backend/src/Fibertest30.Application/SystemEvent/EventData/RtuEvents/RtuAccidentAddedData.cs
using System;
using System.Text.Json;

namespace Fibertest30.Application;

public class RtuAccidentAddedData : ISystemEventData
{
    public int EventId { get; set; } // get from Id
    public DateTime RegisteredAt { get; }
    public string Obj { get; }
    public string ObjId { get; set; }
    public string EventType { get; set; }
    public bool IsGoodAccident { get; set; }

    public RtuAccidentAddedData(int eventId, DateTime registeredAt, string obj, string objId, bool isGoodAccident)
    {
        EventId = eventId;
        RegisteredAt = registeredAt;
        Obj = obj;
        ObjId = objId;
        EventType = "RtuAccident";
        IsGoodAccident = isGoodAccident;
    }

    public string ToJsonData()
    {
        return JsonSerializer.Serialize(this);
    }
}

File: WebFt30/src/app/shared/system-events/system-event-data/rtu-events/rtu-accident-added-data.ts
export interface RtuAccidentAddedData {
  EventId: number;
  RegisteredAt: Date;
  At: string;
  EventType: string;
  Obj: string;
  ObjId: string;
  IsGoodAccident: boolean;
}

export interface NetworkEventAddedData {
  EventId: number;
  RegisteredAt: Date;
  At: string;
  EventType: string;
  Obj: string;
  ObjId: string;
  IsRtuAvailable: boolean;
}

export interface BopNetworkEventAddedData {
  EventId: number;
  RegisteredAt: Date;
  At: string;
  EventType: string;
  Obj: string;
  ObjId: string;
  IsOk: boolean;
}

export interface MeasurementAddedData {
  EventId: number;
  RegisteredAt: Date;
  At: string;
  EventType: string;
  Obj: string;
  ObjId: string;
  IsOk: boolean;
}
